### PR TITLE
TranslateOperation

### PR DESCRIPTION
**Pull request by box-ehuang**

### CODE REVIEW

File: src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    public JsonNode apply(final JsonNode node)
+        throws JsonPatchException
+    {
+        final JsonNode ret = node.deepCopy();
+        if (path.isEmpty()) {
+            if (EQUIVALENCE.equivalent(ret, fromValue)) {
+                return MissingNode.getInstance();
```
Comment (hnguyen08) [inline src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java]: You'd want to return toValue.deepCopy()

File: src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                case SKIP:
+                    return ret;
+            }
+        }
+
+        if (EQUIVALENCE.equivalent(valueAtPath, fromValue)) {
+            final JsonNode parent = path.parent().get(ret);
+            final String rawToken = Iterables.getLast(path).getToken().getRaw();
+            if (parent.isObject())
+                ((ObjectNode) parent).set(rawToken, toValue);
```
Comment (hnguyen08) [inline src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java]: make sure toValue is deepCopy()-ed

File: src/test/resources/jsonpatch/extended/omit.json [hunk 3]
```diff
@@ -63,4 +63,5 @@
             "expected": { "x": "z" }
         }
     ]
-}
\ No newline at end of file
+}
+
```
Comment (hnguyen08) [inline src/test/resources/jsonpatch/extended/omit.json]: Extraneous whitespace

File: src/test/resources/jsonpatch/extended/testsuite.json [hunk 4]
```diff
@@ -11,6 +11,12 @@
         "patch": [{"op": "omit?", "path": "/foo", "value": "bar"}],
         "expected": {}
     },
+  {
+        "comment": "translate operation is able to be parsed and applied",
+        "doc": {"foo": "bar"},
+        "patch": [{"op": "trasnlate", "path": "/foo", "from": "bar", "value": "qux"}],
```
Comment (hnguyen08) [inline src/test/resources/jsonpatch/extended/testsuite.json]: typo -> "translate" (how does this pass?)

File: src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java (hunk 1 above)
Comment (box-ehuang) [inline src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java]: good catch. I copied this when I wasn't sure what handling for empty path was

File: src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java (hunk 2 above)
Comment (box-ehuang) [inline src/main/java/com/github/fge/jsonpatch/operation/TranslateOperationBase.java]: Done

File: src/test/resources/jsonpatch/extended/testsuite.json (hunk 4 above)
Comment (box-ehuang) [inline src/test/resources/jsonpatch/extended/testsuite.json]: It looks like if the op itself is invalid, it skips this test. Maybe exception thrown in the data provider execution? Not sure how the internals of ObjectReader.readValue(...) works on line 64 of JsonPatchTestSuite.java

The entire file in general is being executed though

**Comment by hnguyen08** [comment]
LGTM +2

**Comment by hnguyen08** [comment]
LGTM +2

**Pull request merged**